src/secondary.rs

```diff
@@ -228,4 +245,102 @@ where
 
 impl<'a, K> FusedIterator for BitVecIter<'a, K> where K: TryFrom<usize> {}
 
-// TODO: Implementations for HashSet, BTreeSet, HashMap, BTreeMap.
+impl<K> SecondaryMap<K, bool> for HashSet<K>
+where
+    K: Hash + Eq + Clone,
+{
+    type Iter<'a> = HashSetIter<'a, K> where Self: 'a, K: 'a;
+
+    #[inline]
+    fn new() -> Self {
+        HashSet::new()
+    }
+
+    #[inline]
+    fn with_capacity(capacity: usize) -> Self {
+        HashSet::with_capacity(capacity)
+    }
+
+    #[inline]
+    fn default_value(&self) -> bool {
+        false
+    }
+
+    #[inline]
+    fn ensure_capacity(&mut self, capacity: usize) {
+        HashSet::reserve(self, capacity.saturating_sub(self.capacity()));
+    }
+
+    #[inline]
+    fn resize(&mut self, _new_len: usize) {}
+
+    #[inline]
+    fn capacity(&self) -> usize {
+        HashSet::capacity(self)
+    }
+
+    #[inline]
+    fn get(&self, key: K) -> &bool {
+        if HashSet::contains(self, &key) {
+            &true
+        } else {
+            &false
+        }
```

Comment:
```suggestion
        &HashSet::contains(self, &key)
```

Reply:
Annoyingly we cannot do that, because the return value of `HashSet::contains(self, &key)` is a local variable, where `true` and `false` can be statically defined :/

Reply:
that is quite funny